packages/@aws-cdk/aws-kinesis/lib/stream.ts

```diff
@@ -149,13 +164,7 @@ abstract class StreamBase extends Resource implements IStream {
    * encrypt/decrypt will also be granted.
    */
   public grantReadWrite(grantee: iam.IGrantable) {
-    const ret = this.grant(
-      grantee,
-      'kinesis:DescribeStream',
-      'kinesis:GetRecords',
-      'kinesis:GetShardIterator',
-      'kinesis:PutRecord',
-      'kinesis:PutRecords');
+    const ret = this.grant(grantee, ...Array.from(new Set([...READ_OPERATIONS, ...WRITE_OPERATIONS])));
```

Comment:
This is what we usually do i think: maybe something like this? https://github.com/aws/aws-cdk/blob/master/packages/@aws-cdk/aws-s3/lib/bucket.ts#L524 ?

Reply:
Oh its because of the overlapping action. its ok i guess.